### DISCUSSION TITLE
Adding methods for getting specific neighbors in square grids

### DISCUSSION THREAD

**Discussion by NicolasAmado** [mesa/mesa#1603] [Ideas]
It would be helpfull to add methods for SimpleGrid that gets, for example, the neighbor 1 space to "the right" (It should take into account periodic boundary conditions).

Maybe something like grid.get_single_neighbor(pos, direction, distance=1)
where direction is a string "up", "down", "left", "right" and distance is how many cells in that direction.
for example the method get_single_neighbor((x,y), "right", distance=1) would return the agent in the position (x+1,y) unless it's out of boundaries.

**Comment by rht** [comment]
I have 2 questions:
1. What model do you use this for? I want to know if the use case is commonly encountered
2. Is this canned function available in NetLogo/MASON?

**Reply by NicolasAmado** [reply]
For 1, I was trying to replicate the 2D Sznajd model for Opinion [Dynamics in Effect of Social Media on Opinion Formation](https://arxiv.org/pdf/1805.08310.pdf), by Kamyar Nazeri _(you can check fig. 7)_.
For 2, I can't say, as I haven't used them before. I'm really new with Mesa aswell, I previously made all my agent based simulations by myself.

**Reply by rht** [reply]
I think at this point it makes more sense to implement the function as a part of an example for the Sznajd model in https://github.com/projectmesa/mesa-examples. People can do rapid prototyping to modify the model, e.g adding an external field as it is done in the paper you linked.
